components/properties/src/props.rs

```diff
@@ -685,19 +42,31 @@ macro_rules! create_const_array {
     }
 }
 
-/// Enumerated property Bidi_Class
-///
-/// These are the categories required by the Unicode Bidirectional Algorithm.
-/// For the property values, see [Bidirectional Class Values](https://unicode.org/reports/tr44/#Bidi_Class_Values).
-/// For more information, see [Unicode Standard Annex #9](https://unicode.org/reports/tr41/tr41-28.html#UAX9).
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[cfg_attr(feature = "datagen", derive(databake::Bake))]
-#[cfg_attr(feature = "datagen", databake(path = icu_properties))]
-#[allow(clippy::exhaustive_structs)] // newtype
-#[repr(transparent)]
-#[zerovec::make_ule(BidiClassULE)]
-pub struct BidiClass(pub u8);
+macro_rules! make_map_property {
+    (
+        // currently unused
+        property: $p:expr;
+        // currently unused
+        dyn_marker: $d:ident;
+        value: $value_ty:path;
+        data_marker: $data_marker:ty;
+        func:
+        $(#[$doc:meta])*
+        $vis2:vis const $constname:ident => $singleton:ident;
+        $vis:vis fn $name:ident();
+    ) => {
+        impl crate::private::Sealed for $value_ty {}
+
+        impl crate::code_point_map::CodePointMapProperty for $value_ty {
+            type DataMarker = $data_marker;
+            #[cfg(feature = "compiled_data")]
+            const SINGLETON: &'static crate::provider::PropertyCodePointMapV1<'static, Self> =
+                crate::provider::Baked::$singleton;
+        }
+    };
+}
+
+pub use crate::provider::props::BidiClass;
```

Comment:
issue: I'd rather not do this; these types are tagged as unstable in the docs.

Can we instead just use regular integers in the data and convert here? we erase the type anyway

Reply:
I think we should consider them stable instead.

Reply:
The reason I moved them is that they have `make_ule`, and the ULE types should not be stable.

Reply:
@robertbastian Right, I wish to also consider them stable, which is why I don't want them in the provider module.


Don't worry about `make_ule`, those are going to go away before 1.0 https://github.com/unicode-org/icu4x/issues/5127 . I've been waiting for this PR to land to make that change.